scripts/sct_maths.py

```diff
@@ -505,254 +459,67 @@ def get_data(list_fname):
+    :param type: string: int, float
+    :return:
     """
-    if len(np.shape(data1)) == 3:
-        data1 = data1[..., np.newaxis]
-    if len(np.shape(data2)) == 3:
-        data2 = data2[..., np.newaxis]
-    return np.concatenate((data1, data2), axis=3)
+    new_type_list = (string_list).split(",")
+    for inew_type_list, ele in enumerate(new_type_list):
+        if type is "int":
+            new_type_list[inew_type_list] = int(ele)
+        elif type is "float":
+            new_type_list[inew_type_list] = float(ele)
 
+    return new_type_list
 
-def denoise_nlmeans(data_in, patch_radius=1, block_radius=5):
+def compute_similarity(img1: Image, img2: Image, fname_out: str, metric: str, metric_full: str, verbose):
     """
-    data_in: nd_array to denoise
-    for more info about patch_radius and block radius, please refer to the dipy website: http://nipy.org/dipy/reference/dipy.denoise.html#dipy.denoise.nlmeans.nlmeans
+    Sanitize input and compute similarity metric between two images data.
     """
-    from dipy.denoise.nlmeans import nlmeans
-    from dipy.denoise.noise_estimate import estimate_sigma
-    from numpy import asarray
-    data_in = asarray(data_in)
+    if img1.data.size != img2.data.size:
+        raise ValueError(f"Input images don't have the same size! \nPlease use  \"sct_register_multimodal -i im1.nii.gz -d im2.nii.gz -identity 1\"  to put the input images in the same space")
 
-    block_radius_max = min(data_in.shape) - 1
-    block_radius = block_radius_max if block_radius > block_radius_max else block_radius
-
-    sigma = estimate_sigma(data_in)
-    denoised = nlmeans(data_in, sigma, patch_radius=patch_radius, block_radius=block_radius)
-
-    return denoised
+    if verbose > 1:
+        res = math.compute_similarity(img1.data, img2.data, metric=metric, qc_func=compute_similarity_qc_func)
+    else:
+        res = math.compute_similarity(img1.data, img2.data, metric=metric)
 
+    path_out, filename_out, ext_out = extract_fname(fname_out)
+    if ext_out not in ['.txt', '.pkl', '.pklz', '.pickle']:
+        raise ValueError(f"The output file should a text file or a pickle file. Received extension: {ext_out}")
 
-def smooth(data, sigmas):
-    """
-    Smooth data by convolving Gaussian kernel
-    :param data: input 3D numpy array
-    :param sigmas: Kernel SD in voxel
-    :return:
-    """
-    assert len(data.shape) == len(sigmas)
-    from scipy.ndimage.filters import gaussian_filter
-    return gaussian_filter(data.astype(float), sigmas, order=0, truncate=4.0)
+    if ext_out == '.txt':
+        with open(fname_out, 'w') as f:
+            f.write(metric_full + ': \n' + str(res))
+    elif ext_out == '.pklz':
+        pickle.dump(res, gzip.open(fname_out, 'wb'), protocol=2)
+    else:
+        pickle.dump(res, open(fname_out, 'w'), protocol=2)
 
+def compute_similarity_qc_func(data1_1d, data2_1d, res, metric):
+    import matplotlib
+    matplotlib.use('Agg')
+    import matplotlib.pyplot as plt
```

Comment:
Nit, but could these imports be done at the top of the script?

Reply:
> +1 on these efforts! Feel free to ping me on future refactors. I'd love to help out with reviews. slightly_smiling_face

Thanks for the excellent feedback. I will involve you in all refactor PRs